src/extension.ts

```diff
@@ -607,6 +607,31 @@ function launchMetals(
       }
     );
 
+    registerCommand(`metals.new-scala-worksheet`, async () => {
+      const sendRequest = (args: Array<string | undefined>) => {
+        return client.sendRequest(ExecuteCommandRequest.type, {
+          command: ServerCommands.NewScalaFile,
+          arguments: args,
+        });
+      };
+      const currentUri = window.activeTextEditor?.document.uri;
+      if (currentUri) {
+        const parentUri = path.dirname(currentUri.toString());
+        const name = path.basename(parentUri);
+        const parentPath = Uri.parse(parentUri).fsPath;
+        const fullPath = path.join(parentPath, `${name}.worksheet.sc`);
+        if (fs.existsSync(fullPath)) {
+          return workspace
+            .openTextDocument(fullPath)
+            .then((textDocument) => window.showTextDocument(textDocument));
```

Comment:
Added an option to open the document if it exists.

Reply:
Ah, good idea 👍 

I'll probably add this check into `coc-metals` as well.

Reply:
I'm wondering whether we should also warn the user in this case. Something along the lines of

```ts
window.showWarningMessage(`A worksheet ${name}.worksheet.sc already exists, opening it instead`)
```

Reply:
Added, I think it makes sense. It shouldn't be too annoying as most people would delete the worksheets afterwards